src/BlazorUI/Bit.BlazorUI.Tests/SnackBar/BitSnackBarTests.cs

```diff
@@ -0,0 +1,205 @@
+        DataRow(BitSnackBarType.SevereWarning),
+        DataRow(null)
+    ]
+    [TestMethod]
+    public async Task BitSnackBarTypeTest(BitSnackBarType? type)
+    {
+        var com = RenderComponent<BitSnackBar>();
+
+        if (type.HasValue)
+            await com.Instance.Show("title", string.Empty, type);
+        else
+            await com.Instance.Show("title");
+
+        var element = com.Find(".bit-snb-itm");
+
+        var typeClass = type switch
+        {
+            BitSnackBarType.Info => $"bit-snb-info",
+            BitSnackBarType.Warning => $"bit-snb-warning",
+            BitSnackBarType.Success => $"bit-snb-success",
+            BitSnackBarType.Error => $"bit-snb-error",
+            BitSnackBarType.SevereWarning => $"bit-snb-severe-warning",
+            _ => "bit-snb-info"
+        };
+
+        Assert.IsTrue(element.ClassList.Contains(typeClass));
+    }
+
+    [TestMethod]
+    public async Task BitSnackBarCloseButtonTest()
+    {
+        var com = RenderComponent<BitSnackBar>();
+
+        await com.Instance.Show("title");
+
+        var closeButton = com.Find(".bit-snb-cbt");
+        closeButton.Click();
+
+        var items = com.FindAll(".bit-snb-itm");
+
+        Assert.AreEqual(0, items.Count);
+    }
+
+    [TestMethod]
+    public async Task BitSnackBarDismissIconNameTest()
+    {
+        var com = RenderComponent<BitSnackBar>(
+            parameters =>
+            {
+                parameters.Add(x => x.DismissIconName, "Go");
+            }
+        );
+
+        await com.Instance.Show("title");
+
+        var closeButtonIcon = com.Find(".bit-snb-cbt > .bit-icon");
+
+        Assert.IsTrue(closeButtonIcon.ClassList.Contains("bit-icon--Go"));
+    }
+
+    [TestMethod]
+    public async Task BitSnackBarTitleTemplateTest()
+    {
+        var wrappedCom = RenderComponent<BitSnackBarWithTitleTemplate>();
+
+        var com = wrappedCom.FindComponent<BitSnackBar>();
+
+        await com.Instance.Show("title");
+
+        var titleTemplateElements = com.FindAll(".bit-snb-hdr > *").ToList();
+        //remove button
+        titleTemplateElements.RemoveAt(0);
+
+        Assert.AreEqual("<span>title</span><span>second element</span>", string.Join(string.Empty, titleTemplateElements.Select(x => x.OuterHtml).ToArray()));
+    }
+
+    [TestMethod]
+    public async Task BitSnackBarBodyTemplateTest()
+    {
+        var wrappedCom = RenderComponent<BitSnackBarWithBodyTemplate>();
+
+        var com = wrappedCom.FindComponent<BitSnackBar>();
+
+        await com.Instance.Show("title", "body");
+
+        var itemTemplateElements = com.FindAll(".bit-snb-itm > *").ToList();
+        //remove title
+        itemTemplateElements.RemoveAt(0);
+        //remove progressbar
+        itemTemplateElements.RemoveAt(itemTemplateElements.Count - 1);
+
+        Assert.AreEqual("<p>body</p><span>second element</span>", string.Join(string.Empty, itemTemplateElements.Select(x => x.OuterHtml).ToArray()));
```

Comment:
again, parent InnerHtml instead of children's OuterHtml.

Reply:
same as the above problem, here we have an progressbar in additional

Reply:
again:

https://bunit.dev/docs/verification/verify-component-state.html
https://bunit.dev/docs/verification/semantic-html-comparison.html